### ISSUE TITLE
TestVerifyOdsQuickstarterProvisionThruProvisionApi fails due to repository URL already taken

### ISSUE THREAD

**Issue by renedupont** [opendevstack/ods-quickstarters#435]
**Describe the bug**

While running the test ´TestVerifyOdsQuickstarterProvisionThruProvisionApi ´ the following error occured:

```
An error occured while updating project ODSVERIFY, reason Error in creating repo: releasemanager
        details: Could not POST > https://bitbucket.<our_domain>.com/rest/api/1.0/projects/ODSVERIFY/repos : {"errors":[{"context":"name","message":"This repository URL is already taken.","exceptionName":null}]} Errorcode: 409 - but all cleaned up!
        ... displaying HTTP response code
        http_resp_code=500
```

I guess that relates to that code fragment: https://github.com/opendevstack/ods-quickstarters/blob/master/tests/scripts/setup_bitbucket_test_project.sh#L131
The mentioned repo has just been deleted a few lines above... is it possible that we have to wait for a few sec until it finished the deletion?

Will try soon to add a sleep if I don't get logged out by openshift before it comes to that (very last) test ...

This is a ODS 3.0 release blocker.

fyi @michaelsauter @clemensutschig

**Comment by clemensutschig** [comment]
Hmmm ... we dont delete repos on purpose..  so i think we need to extend the Test to Call the bitbucket api to delete it.. @michaelsauter

**Comment by clemensutschig** [comment]
@renedupont can u Paste a Screenshot from bitbucket odsverify Project?

**Comment by renedupont** [comment]
here is more debug from above it:

```
create or update project, or delete component - DELETE_COMPONENT
... DELETE_COMPONENT project request payload loaded from 'golden/create-quickstarter-request.json'

... displaying payload file content:
{
  "projectKey" : "ODSVERIFY",
  "quickstarters" : [
	  {
	    "component_type" : "release-manager",
	    "component_id" : "releasemanager"
	  },
	  {
	    "component_type" : "be-golang-plain",
	    "component_id" : "golang"
	  }
  ]
}
curl DELETE request successful...

... displaying HTTP response body (content from './response.txt'):
<some secret data...>
... displaying HTTP response code
http_resp_code=200

provision project request (DELETE) completed successfully!!!

--- FAIL: TestVerifyOdsQuickstarterProvisionThruProvisionApi (38.91s)
    jenkinsfile_test.go:93: Execution of `create-project-api.sh` failed: 
        StdOut: No config file found, assuming defaults, current dir: /c/Users/dupontre/IdeaProjects/opendevstack/ods-quickstarters/tests/release-manager
        
        Started provision project script with command (PUT)!
        
        ... encoding basic auth credentials in base64 format
        
        ... sending request to 'https://<prov_app_url>' (output will be saved in file './response.txt' and headers in file './headers.txt')
        
        
        create or update project, or delete component - PUT
        ... PUT project request payload loaded from 'golden/create-quickstarter-request.json'
        
        ... displaying payload file content:
        {
          "projectKey" : "ODSVERIFY",
          "quickstarters" : [
        	  {
        	    "component_type" : "release-manager",
        	    "component_id" : "releasemanager"
        	  },
        	  {
        	    "component_type" : "be-golang-plain",
        	    "component_id" : "golang"
        	  }
          ]
        }
        curl PUT request successful...
        
        ... displaying HTTP response body (content from './response.txt'):
        An error occured while updating project ODSVERIFY, reason Error in creating repo: releasemanager
        details: Could not POST > https://bitbucket.<our_domain>.com/rest/api/1.0/projects/ODSVERIFY/repos : {"errors":[{"context":"name","message":"This repository URL is already taken.","exceptionName":null}]} Errorcode: 409 - but all cleaned up!
        ... displaying HTTP response code
        http_resp_code=500
        
        something went wrong... endpoint responded with error code [HTTP CODE=500] (expected was 200)
```

**Comment by renedupont** [comment]
giving jenkins cd user admin rights solved the issue above, but the test still fails due to:

--- FAIL: TestVerifyOdsQuickstarterProvisionThruProvisionApi (622.84s)
Error: exit status 4, , parse error: Invalid literal at line 1, column 79512out: '', err: Error getting jenkins stages for: mro-pipeline-master-2

@clemensutschig

**Comment by clemensutschig** [comment]
Boahhh this is insanely painfull.. still no Solution - and an Error prone Workaround..  See: https://github.com/opendevstack/ods-quickstarters/blob/master/tests/scripts/print-jenkins-json-status.sh#L3 Maybe @michaelsauter  you have better idea?

**Comment by renedupont** [comment]
btw, I get the same for the ds-ml-service test, I guess it has the same origin then?

```
--- FAIL: TestJenkinsFile (526.14s)
    jenkinsfile_test.go:89: Error getting jenkins stages for: run-ml-unitt-master-13
Error: exit status 4, , parse error: Invalid numeric literal at line 1, column 20425
        
FAIL
FAIL	github.com/opendevstack/ods-quickstarters/tests/ds-ml-service	526.717s
```

**Comment by michaelsauter** [comment]
I don't know why we are seeing those errors. I just tried to get the information differently and it works:
```
oc get build run-ml-unitt-master-13 -n unitt-cd -ojsonpath='{.metadata.annotations.openshift\.io/jenkins-status-json}' | jq '[.stages[] | {stage: .name, status: .status}]'                                                                                             [
  {
    "stage": "odsPipeline start",
    "status": "SUCCESS"
  },
  {
    "stage": "Lint",
    "status": "SUCCESS"
  },
  {
    "stage": "Prepare Builds",
    "status": "SUCCESS"
  },
  {
    "stage": "Build OpenShift Image (ml-training-service)",
    "status": "SUCCESS"
  },
  {
    "stage": "Deploy to OpenShift (ml-training-service)",
    "status": "SUCCESS"
  },
  {
    "stage": "Training Unit Tests",
    "status": "SUCCESS"
  },
  {
    "stage": "Train Model",
    "status": "SUCCESS"
  },
  {
    "stage": "Training Integration Tests",
    "status": "SUCCESS"
  },
  {
    "stage": "SonarQube Analysis",
    "status": "SUCCESS"
  },
  {
    "stage": "Build OpenShift Image (ml-prediction-service)",
    "status": "SUCCESS"
  },
  {
    "stage": "Deploy to OpenShift (ml-prediction-service)",
    "status": "SUCCESS"
  },
  {
    "stage": "odsPipeline finished",
    "status": "SUCCESS"
  }
]
```

I need to investigate why this workaround is there - it looks like it is not needed when we get the annotation directly?

**Comment by clemensutschig** [comment]
Nice. Therme is no reason... it das the only way i could make it work ;-)

Michael Sauter <notifications@github.com> schrieb am Mi., 5. Aug. 2020,
15:56:

**Comment by clemensutschig** [comment]
@michaelsauter no reasons.. that current way in print-jenkins-json-status was the only way i could make it work.. feel free to Change for better.. ;-)

**Comment by michaelsauter** [comment]
@clemensutschig OK then I will do some checks tomorrow to see if I can change to how I extracted the info...

**Comment by michaelsauter** [comment]
Fixed in #437.
